xlators/features/locks/src/posix.c

```diff
@@ -207,24 +216,24 @@ dict_delete_domain_key(dict_t *dict, char *key, data_t *value, void *data)
     return 0;
 }
 
-void
-pl_get_xdata_requests(pl_local_t *local, dict_t *xdata)
+static void
+pl_get_xdata_requests(pl_local_t *local, dict_t *xdata, unsigned char bitfield)
 {
     if (!local || !xdata)
         return;
 
     GF_ASSERT(local->xdata == NULL);
     local->xdata = dict_copy_with_ref(xdata, NULL);
 
-    if (dict_get_sizen(xdata, GLUSTERFS_ENTRYLK_COUNT)) {
+    if (IS_BIT_SET(bitfield, 0)) {
```

Comment:
It would be much better (and safe) to use macros or an enum instead of constants. And use the same macros/enums when initializing the `reqs` and `reqs_size` arrays in line 182 (like `[GLUSTERFS_ENTRYLK_COUNT_BIT] = GLUSTERFS_ENTRYLK_COUNT`)